deepmd/pt/model/task/denoise.py

```
@@ -1,79 +1,295 @@
-            self.pair2coord_proj = torch.nn.ModuleList(self.pair2coord_proj)
+                for ii in range(self.ntypes if not self.mixed_types else 1)
+            ],
+        )
+
+        # set trainable
+        for param in self.parameters():
+            param.requires_grad = self.trainable
+
+    def reinit_exclude(
+        self,
+        exclude_types: list[int] = [],
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)